src/common/transformations/src/transformations/common_optimizations/softmax_fusion.cpp

```diff
@@ -8,63 +8,74 @@
 #include <ngraph/pattern/op/wrap_type.hpp>
 #include <ngraph/rt_info.hpp>
 #include <openvino/opsets/opset6.hpp>
+#include <openvino/pass/pattern/op/or.hpp>
 #include <vector>
 
 #include "itt.hpp"
 #include "transformations/utils/utils.hpp"
 
 ov::pass::SoftmaxFusion::SoftmaxFusion() {
     MATCHER_SCOPE(SoftmaxFusion);
-    auto data_pattern = pass::pattern::any_input(pattern::has_static_rank());
+
+    auto data_pattern = pass::pattern::any_input(pass::pattern::has_static_rank());
     auto reduce_max_axes_pattern = ngraph::pattern::wrap_type<opset6::Constant>();
     auto reduce_max_pattern = ngraph::pattern::wrap_type<opset6::ReduceMax>({data_pattern, reduce_max_axes_pattern});
     auto sub_pattern = ngraph::pattern::wrap_type<opset6::Subtract>({data_pattern, reduce_max_pattern});
-    auto exp_pattern = ngraph::pattern::wrap_type<opset6::Exp>({sub_pattern});
+
+    auto exp_input = std::make_shared<pattern::op::Or>(OutputVector{sub_pattern, data_pattern});
+    auto exp_pattern = ngraph::pattern::wrap_type<opset6::Exp>({exp_input});
+
     auto reduce_sum_axes_pattern = ngraph::pattern::wrap_type<opset6::Constant>();
     auto reduce_sum_pattern = ngraph::pattern::wrap_type<opset6::ReduceSum>({exp_pattern, reduce_sum_axes_pattern});
     auto div_pattern = ngraph::pattern::wrap_type<opset6::Divide>({exp_pattern, reduce_sum_pattern});
 
-    ov::matcher_pass_callback callback = [=](pattern::Matcher& m) {
+    ov::matcher_pass_callback callback = [=](pass::pattern::Matcher& m) {
         if (transformation_callback(m.get_match_root()))
             return false;
 
         const auto& pattern_map = m.get_pattern_value_map();
+        const auto& pshape = pattern_map.at(data_pattern).get_partial_shape();
+        auto rank = pshape.rank().get_length();
 
-        auto reduce_max_axes =
-            std::dynamic_pointer_cast<opset6::Constant>(pattern_map.at(reduce_max_axes_pattern).get_node_shared_ptr());
-        if (!reduce_max_axes || shape_size(reduce_max_axes->get_shape()) != 1)
-            return false;
         auto reduce_sum_axes =
             std::dynamic_pointer_cast<opset6::Constant>(pattern_map.at(reduce_sum_axes_pattern).get_node_shared_ptr());
         if (!reduce_sum_axes || shape_size(reduce_sum_axes->get_shape()) != 1)
             return false;
-
-        int64_t reduce_max_axis = reduce_max_axes->cast_vector<int64_t>()[0];
         int64_t reduce_sum_axis = reduce_sum_axes->cast_vector<int64_t>()[0];
-        if (reduce_max_axis < 0 || reduce_sum_axis < 0) {
-            const auto& pshape = pattern_map.at(data_pattern).get_partial_shape();
-            auto rank = pshape.rank().get_length();
```

Comment:
Rank is only needed when any of axes are negative, could you do that too?

Reply:
Done. Now get the rank is obtained only when it's needed (axis < 0).